File: GitToolBox/src/main/java/zielu/gittoolbox/metrics/ProjectMetrics.java
package zielu.gittoolbox.metrics;

import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.NotNull;
import zielu.gittoolbox.util.AppUtil;

public interface ProjectMetrics extends Metrics {

  static Metrics getInstance(@NotNull Project project) {
    return AppUtil.getServiceInstance(project, ProjectMetrics.class);
  }
}

File: GitToolBox/src/main/kotlin/zielu/gittoolbox/metrics/ProjectMetrics.kt
package zielu.gittoolbox.metrics

import com.intellij.openapi.project.Project
import zielu.gittoolbox.util.AppUtil.getServiceInstance
import zielu.gittoolbox.util.AppUtil.getServiceInstanceSafe

internal interface ProjectMetrics : Metrics {
  fun startReporting()

  companion object {
    @JvmStatic
    fun getInstance(project: Project): Metrics {
      return getServiceInstance(project, ProjectMetrics::class.java)
    }

    fun startReporting(project: Project) {
      getServiceInstanceSafe(project, ProjectMetrics::class.java).ifPresent { it.startReporting() }
    }
  }
}
